tests/components/samsungtv/test_config_flow.py

```diff
@@ -2018,15 +1994,17 @@ async def test_form_reauth_encrypted(hass: HomeAssistant) -> None:
     del encrypted_entry_data[CONF_TOKEN]
     del encrypted_entry_data[CONF_SESSION_ID]
 
-    entry = await setup_samsungtv_entry(hass, encrypted_entry_data)
-    assert entry.state == config_entries.ConfigEntryState.SETUP_ERROR
-    flows_in_progress = [
-        flow
-        for flow in hass.config_entries.flow.async_progress()
-        if flow["context"]["source"] == "reauth"
-    ]
-    assert len(flows_in_progress) == 1
-    result = flows_in_progress[0]
+    entry = MockConfigEntry(domain=DOMAIN, data=encrypted_entry_data)
+    entry.add_to_hass(hass)
+    assert entry.state == config_entries.ConfigEntryState.NOT_LOADED
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"entry_id": entry.entry_id, "source": config_entries.SOURCE_REAUTH},
+        data=entry.data,
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
```

Comment:
Now that setup is mocked, we need to initialise the reauth flow manually.

Reply:
Doesn't this change make the test worse? Or is there a test in `test_init.py` which ensure reauth is triggered when needed?

Reply:
Coverage is still 100%.
The reauth tests are in `test_media_player.py`

Reply:
Anyway - I added a specific test in 9d2d87dfd62ca5ee0f2c882b1289d874008e9960